src/generate.ts

```diff
@@ -0,0 +1,385 @@
+      const currValue = typeMap.get(key)!;
+      if ((type as ComplexTypeNodeWithComplexContent).complexContent) {
+        // complex content case
+        const t = type as ComplexTypeNodeWithComplexContent;
+        currValue.parents.push(t.complexContent.extension.$.base);
+        currValue.fields.push(...(toArray(t.complexContent.extension.sequence?.element) as SequenceNode[]));
+      } else {
+        currValue.fields.concat(
+          ((type as ComplexTypeNodeWithSequence).sequence?.element as SequenceNode[]) ??
+            (type as SimpleTypeNode).restriction.enumeration
+        );
+      }
+      typeMap.set(key, currValue);
+    }
+  });
+
+  typeMap.forEach((info, type) => {
+    if (info.fields.every((f) => 'value' in f.$)) {
+      // we have an enum type
+      output += 'export type ' + type + ' = ';
+      if (info.fields.length === 0) output += 'string';
+      output += info.fields.map((f) => "'" + f.$.value + "'").join('\n      |');
+      output += '\n\n';
+    } else {
+      output += 'export type ' + type + ' = ';
+      output += info.parents.map((p) => treatTypeName(p)).join(' & ');
+      if (info.parents.length) {
+        output += ' & {\n';
+      } else {
+        output += '{\n';
+      }
+      // there's a duplciate field on UIObjectRelationConfig - with potentially more to come, filter out the duplicate
+      // as well as fields that shouldn't be here !('value' in n.$')
+      info.fields
+        .filter((n, i) => info.fields.map((f) => f.$.name).indexOf(n.$.name) === i && !('value' in n.$))
+        .map((f) => {
+          output += treatAttribute(f);
+        });
+      output += '}\n\n';
+    }
+  });
+
+  // write the ApiSchemaTypes type - everything below this can be removed if we deem it not necessary
+  // this is something jsforce was doing, that might make the transition away from types in jsforce easier
+
+  output += 'export type ApiSchemaTypes = {\n';
+
+  typeMap.forEach((info, type) => {
+    output += `\t${type}: ${type};\n`;
+  });
+  output += '}\n';
+
+  // remove to here
+
+  return output;
+}
+
+function treatAttribute(elementNode: NodeWithAttributes): string {
+  let attributeOutput = '';
+
+  const { name: fieldName, type: fieldTypeXml, minOccurs, maxOccurs, nillable } = elementNode.$;
+
+  console.assert(fieldName !== undefined, 'A field name is undefined', elementNode.$);
+  console.assert(fieldTypeXml !== undefined, 'A field type is undefined');
+
+  const isArray = maxOccurs === 'unbounded';
+  const optional = (!isArray && minOccurs === '0') || nillable ? '?' : '';
+  attributeOutput += '    ' + fieldName + optional + ': ';
+
+  const fieldType = translateTypeName(fieldTypeXml ?? '') ?? 'any';
+
+  if (isArray) {
+    attributeOutput += fieldType + '[]';
+  } else {
+    attributeOutput += fieldType;
+  }
+
+  attributeOutput += ';\n';
+
+  return attributeOutput;
+}
+
+function translateTypeName(fieldTypeXml: string): string {
+  let fieldType = '';
```

Comment:
you can `return` from a switch/case and not have to have all the `break;`, the `let` and the final `return` of your var